packages/core/src/dialects/ibmi/query-generator-typescript.ts

```diff
@@ -217,4 +219,31 @@ export class IBMiQueryGeneratorTypeScript extends AbstractQueryGenerator {
       table.schema ? this.escape(table.schema) : 'CURRENT SCHEMA',
     ]);
   }
+
+  updateQuery(
+    tableOrModel: TableOrModel,
+    attrValueHash: Record<string, unknown>,
+    options?: UpdateQueryOptions,
+  ): QueryWithBindParams {
+    const updateQuery = super.updateQuery(tableOrModel, attrValueHash, { ...options, returning: false });
+
+    if (options?.returning) {
```

Comment:
Following my comment about normalizing `returning` first: It will solve another bug here where if I set `returning` to an empty array, the current code will try to return everything, when I'm actually asking to return nothing.

After normalizing, these conditions should be updated to become `normalizedReturning.length !== 0`

Reply:
I moved all the logic to the `getDb2IbmiSelectFromFinalTable` function